File: solutions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        res = -1

        for r in range (l, len(prices)):
            if(prices[l] > prices[r]):
                l = r
            res = max(res, prices[r] - prices[l])

        return res        

File: solutions/0121-best-time-to-buy-and-sell-stock/solution.ts
function maxProfit(prices: number[]): number {
    // two pointers
    // keep track of currProfit
    // maxProfit = max(currProfit, maxProfit)

    let maxProfit = 0
    let i = 0
    let j = 1

    while (j < prices.length) {
        if (prices[i] > prices[j]) {
            i = j
            j = i + 1
        } else {
            maxProfit = Math.max(maxProfit, prices[j] - prices[i])
            j++
        }
    }

    return maxProfit
};
